### PR TITLE
feat(xmldsig): add VerifyContext builder policies

### LINKED ISSUE #31: feat: implement VerifyContext builder (P1-022)
## Summary
Implement ROADMAP task P1-022:  builder with key resolution and verification policy controls.

## Scope
- Add  builder API for XMLDSig verification
- Support pre-set key and  trait based key resolution
- Add policy controls: , , 
- Keep current verification path compatible while enabling context-driven checks
- Add/extend tests for builder defaults and policy enforcement behavior

## Acceptance Criteria
-  can be constructed via builder with sane defaults
- Verification pipeline can run using explicit key and resolver path
- Transform and URI policy controls are enforced
- Manifest processing flag is wired and covered by tests
-  and  pass

## Estimate
2d

### PR DESCRIPTION
**Pull request by polaz**
## Summary
- add VerifyContext builder with pre-set key and resolver hooks
- add policy controls for allowed URI classes and transform allowlist (including implicit default C14N handling)
- keep verify_signature_with_pem_key API and route it through the new context path
- allow KeyResolver to return non-`'static` verifier objects tied to resolver lifetime
- reject nested `<ds:Manifest>` under `<ds:Object>` when `process_manifests(true)` is enabled
- defer key resolver execution until the final cryptographic verification step
- add regression tests for policy and manifest edge cases

## Testing
- cargo fmt --all --check
- cargo clippy --all-targets -- -D warnings
- cargo nextest run --workspace
- cargo test --doc --workspace
- cargo build --workspace

Closes #31

### CODE REVIEW

File: src/xmldsig/verify.rs [hunk 1]
```diff
@@ -318,6 +538,81 @@ pub fn verify_signature_with_pem_key(
...
+                if !allowed.contains(transform_uri) {
+                    return Err(SignatureVerificationPipelineError::DisallowedTransform {
+                        algorithm: transform_uri.to_owned(),
+                    });
+                }
+            }
+        }
+    }
+    Ok(())
+}
```
[the remark this answers is not in the document]
Comment (polaz) [reply src/xmldsig/verify.rs]: @coderabbitai Fixed. `enforce_reference_policies()` now rejects `Reference.uri == None` with `SignatureVerificationPipelineError::Reference(ReferenceProcessingError::MissingUri)` before URI-type checks, aligning policy and processing behavior. Added regression: `enforce_reference_policies_rejects_missing_uri_before_uri_type_checks`.

File: src/xmldsig/verify.rs [hunk 2]
```diff
@@ -508,6 +803,78 @@ mod tests {
...
+        let err = VerifyContext::new()
+            .key(&RejectingKey)
+            .allowed_transforms(["http://www.w3.org/2001/10/xml-exc-c14n#"])
+            .verify(&xml)
+            .expect_err("enveloped transform should be rejected by allowlist");
+        assert!(matches!(
+            err,
+            SignatureVerificationPipelineError::DisallowedTransform { .. }
+        ));
+    }
```
[the remark this answers is not in the document]
Comment (polaz) [reply src/xmldsig/verify.rs]: @coderabbitai Added. Test `verify_context_manifest_policy_toggle_is_enforced` now asserts `process_manifests(true)` returns `ManifestProcessingUnsupported` and `process_manifests(false)` preserves prior verification flow.

File: src/xmldsig/transforms.rs [hunk 3]
```diff
@@ -26,7 +26,10 @@ use crate::c14n::{self, C14nAlgorithm};
 /// The algorithm URI for the enveloped signature transform.
 pub const ENVELOPED_SIGNATURE_URI: &str = "http://www.w3.org/2000/09/xmldsig#enveloped-signature";
 /// The algorithm URI for the XPath 1.0 transform.
-const XPATH_URI: &str = "http://www.w3.org/TR/1999/REC-xpath-19991116";
+pub const XPATH_TRANSFORM_URI: &str = "http://www.w3.org/TR/1999/REC-xpath-19991116";
+/// The implicit default canonicalization URI applied when no explicit C14N
+/// transform is present in a `<Reference>`.
+pub const DEFAULT_IMPLICIT_C14N_URI: &str = "http://www.w3.org/TR/2001/REC-xml-c14n-20010315";
```
[the remark this answers is not in the document]
Comment (polaz) [reply src/xmldsig/transforms.rs]: @coderabbitai Fixed. Re-exported `DEFAULT_IMPLICIT_C14N_URI`, `ENVELOPED_SIGNATURE_URI`, and `XPATH_TRANSFORM_URI` from `xmldsig::mod` so downstream callers can configure `VerifyContext::allowed_transforms()` without copy-pasting literals.
